### ISSUE TITLE
[Accessibility] Improve the reading of the top items on the home screen with VoiceOver

### ISSUE THREAD

**Issue by mozillamobilebot** [mozilla-mobile/firefox-ios#4991]
From https://bugzilla.mozilla.org/show_bug.cgi?id=1453911

STR:

1. Run Firefox 11 for iOS and create a new tab so th Home screen is shown.
1. Turn on VoiceOver.
1. Touch or swipe through the items right and left.

Expected: VoiceOver should read something useful, like what site is being previewed or would be opened if I double-tapped.

Actual: I only hear things like "Google", "IRCCloud", "Slack", which seem to be parts of the domain names from the URLs. The AccessibilityLabel properties for these should be more comprehensive.

Note that other tabs like Bookmarks, History, and Reading List are not suffering from this, they read much better.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-2112)

**Comment by nbhasin2** [comment]
This is how we currently handle it, if there area any specific A11Y changes required feel free to a new bug or mention in detail for us to look into.
